openpype/hosts/maya/plugins/publish/extract_maya_scene_raw.py

```diff
@@ -64,17 +169,37 @@ def process(self, instance):
 
         # Perform extraction
         self.log.debug("Performing extraction ...")
-        with maintained_selection():
+
+        frame_offset = instance.data.get("frameOffset")
+        self.log.debug("frameOffset: {}".format(frame_offset))
+
+        frame_range = [None, None, None, None]
+        if instance.data.get("update_timeline", False):
+            frame_range = [
+                instance.data["frameStart"],
+                instance.data["frameEnd"],
+                instance.data["handleStart"],
+                instance.data["handleEnd"]
+            ]
+            self.log.debug("Updating timeline to {}.".format(frame_range))
+
+        kwargs = {
+            "force": True,
+            "typ": "mayaAscii" if self.scene_type == "ma" else "mayaBinary",
+            "exportSelected": True,
+            "preserveReferences": True,
+            "constructionHistory": True,
+            "shader": True,
+            "constraints": True,
+            "expressions": True
+        }
+        self.log.debug("Exporting with:\n{}".format(kwargs))
+        with (maintained_selection(),
+              offset_scene(frame_offset),
+              maintain_timeline(*frame_range)
+              ):
             cmds.select(selection, noExpand=True)
-            cmds.file(path,
-                      force=True,
-                      typ="mayaAscii" if self.scene_type == "ma" else "mayaBinary",  # noqa: E501
-                      exportSelected=True,
-                      preserveReferences=True,
-                      constructionHistory=True,
-                      shader=True,
-                      constraints=True,
-                      expressions=True)
+            cmds.file(path, **kwargs)
```

Comment:
Why are these refactored to `kwargs` if they are not dynamic? This doesn't improve readability?